File: front/fixtures/product.js
const product =
{
  id: 1,
  name: '만두 지갑',
  originalPrice: 50000,
  discountedPrice: 40000,
  description: '설명',
  images: [{
    id: 1,
    url: 'url1',
  }, {
    id: 2,
    url: 'url2',
  }],
  options: [
    {
      id: 1,
      name: '색상',
      additionalPrice: 0,
      children: [
        {
          id: 10,
          name: '갈색',
          additionalPrice: 1000,
          children: [],
        },
        {
          id: 10,
          name: '남색',
          additionalPrice: 2000,
          children: [],
        },
      ],
    },
  ],
};

export default product;

File: src/main/java/com/solebysole/product/domain/Product.java
package com.solebysole.product.domain;

import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.Lob;
import javax.persistence.OneToMany;
import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

/**
 * 상품.
 */
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Entity
public class Product {

    @Id
    @GeneratedValue
    @Column(name = "product_id")
    private Long id;

    private String name;

    private int originalPrice;

    private int discountedPrice;

    @Lob
    private String description;

    @Enumerated(EnumType.STRING)
    private Category category;

    @OneToMany(mappedBy = "product", cascade = CascadeType.ALL)
    private Set<Keyword> keywords = new HashSet<>();

    @OneToMany(mappedBy = "product", cascade = CascadeType.ALL)
    private List<Image> images = new ArrayList<>();

    @OneToMany(mappedBy = "product", cascade = CascadeType.ALL)
    private List<Option> options = new ArrayList<>();

    @Builder
    private Product(Long id, String name, int originalPrice, int discountedPrice,
                    String description, Category category) {
        this.id = id;
        this.name = name;
        this.originalPrice = originalPrice;
        this.discountedPrice = discountedPrice;
        this.description = description;
        this.category = category;
    }

    public void addKeyword(Keyword keyword) {
        keywords.add(keyword);
        keyword.setProduct(this);
    }

    public void addImage(Image image) {
        images.add(image);
        image.setProduct(this);
    }

    public void addOption(Option option) {
        options.add(option);
        option.setProduct(this);
    }

}
